spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/AbstractAppRegistryCommon.java

```diff
@@ -56,8 +77,14 @@ public Resource getAppResource(AppRegistration appRegistration) {
 
 	@Override
 	public Resource getAppMetadataResource(AppRegistration appRegistration) {
-		return appRegistration.getMetadataUri() != null ? this.resourceLoader.getResource(
-				appRegistration.getMetadataUri().toString()) : null;
+		if (appRegistration.getMetadataUri() != null) {
+			return this.resourceLoader.getResource(appRegistration.getMetadataUri().toString());
+		}
+		else {
+			appRegistrationResourceCache.putIfAbsent(appRegistration.getUri(),
```

Comment:
I think we want to return `null` in this case where the metadata URI is not set for the app registration.

For instance, from what is reported in #1983 the metadata should have been registered as part of [this](https://github.com/spring-cloud-stream-app-starters/app-starters-release/blob/master/spring-cloud-stream-app-descriptor/src/main/resources/META-INF/rabbit-apps-maven.properties#L90)

Maybe in this case of #1983 the app was registered with `app register` and the metadata wasn't registered at all?

On a related note, this fix would cause the actual app resource to be downloaded(via `getAppResource`) which we want to avoid doing at the server level.

Reply:
The original issue didn't show the registration command.  It did not explicitly list `--metadata-uri`.

This PR shows the same behavior as 1.2.3

```dataflow:>app register --name log --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.2.0.RELEASE --force
Successfully registered application 'sink:log'
dataflow:>app info sink:log
Information about sink application 'log':
Version: '1.2.0.RELEASE':
Default application version: 'true':
Resource URI: maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.2.0.RELEASE
╔══════════════════════════════╤══════════════════════════════╤══════════════════════════════╤══════════════════════════════╗
║         Option Name          │         Description          │           Default            │             Type             ║
╠══════════════════════════════╪══════════════════════════════╪══════════════════════════════╪══════════════════════════════╣
║log.name                      │The name of the logger to use.│<none>                        │java.lang.String              ║
║log.level                     │The level at which to log     │<none>                        │org.springframework.integratio║
║                              │messages.                     │                              │n.handler.LoggingHandler$Level║
║log.expression                │A SpEL expression (against the│payload                       │java.lang.String              ║
║                              │incoming message) to evaluate │                              │                              ║
║                              │as the logged message.        │                              │                              ║
╚══════════════════════════════╧══════════════════════════════╧══════════════════════════════╧══════════════════════════════╝
```